python/pyspark/sql/functions.py

```diff
@@ -2548,6 +2549,10 @@ def pandas_udf(f=None, returnType=None, functionType=None):
        |  2|6.0|
        +---+---+
 
+       .. note:: If returning a new `pandas.DataFrame` constructed with a dictionary, it is
+       recommended to explicitly index the columns by name to ensure the positions are correct.
+       For example, `pd.DataFrame({'id': ids, 'a': data}, columns=['id', 'a'])`.
```

Comment:
Should we recommend OrderedDict? I feel specifying column names twice is not ideal